### PR TITLE
Using file basename as namespace

### PR DESCRIPTION
**Pull request by bradpe**
i18next wants a namespace in the resources.

### CODE REVIEW

**Comment by rosskevin** [comment]
I specifically left the filename separate from content I order to allow the user to put on or many locales in a single file.  Generally I use one but there are some cases in our transition where it would simply mean an explosion of files for one or two i18n keys.

Is there a compelling use case to impose a restriction?

**Comment by brad-peters** [comment]
When I pass the resources, as in the example, in to i18next it can't find them unless the vales in the files are nested inside a namespace object.

**Comment by rosskevin** [comment]
Yes, exactly.  [It was my intent](https://github.com/atroo/i18next-resource-store-loader/issues/14#issuecomment-331726268) to allow the locale author to specify the key under which to namespace the values, preventing an unnecessary binding to filenames causing a multiplication of needed files.

e.g.  `contact.yml` has the keys for multiple components `ContactEdit`, `ContactAdd` and `ContactList`:

```yaml
ContactEdit:
  success: Update successful
  failure: Update failed

ContactAdd:
  success: Create successful
  failure: Create failed

ContactList:
  title: Contact List
  success: Update successful
  failure: Update failed
```

In ^^^ case, I don't want three files for the area of `contact`.

But I will say there are cases where we do use one file per component - it just depends on the amount of content.

I'm open to either:
1.  A heuristic that chooses to use the `basename` as namespace when starts-with uppercase letter e.g. for complex components with more content - they have their own files like `AccessActivated.yml`
2. An option passed to use the `basename` as the top level namespace
3. Pluggable option/function that allows you to determine your own namespace

The default should be no namespace - it is provided by the file.

**Review by rosskevin — CHANGES_REQUESTED** [review]
Changes requested are outlined in the general discussion.

**Comment by bradpe** [comment]
Ok I think I understand now.  

I'm using i18next-parser and it uses the namespace as the filename.
So my t("key.subkey") gets extracted to "en/translation.json" as
``` json
{ 
  "key": {
    "subkey": ""
  }
}
```
If it extracted as:
``` json
{ 
  "translation": {
    "key": {
      "subkey": ""
    }
  }
}
```
Then your loader would work as expected and the filenames don't matter

**Comment by rosskevin** [comment]
Great, so are you still proposing a change or should we close this out?

**Comment by brad-peters** [comment]
I'll take a look at their code. But I think I like the use  basename option best. I'll implement that and update.

**Comment by rosskevin** [comment]
Please add:
- `basename.spec.js` with a simple test to ensure you get the proper top level key/data
- README section with your new `Advanced usage` option

File: test/basename.spec.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          assert.notInclude(path, 'exclude.json')
+        }
+
+        const res = loader.call(thisScope, 'index.js')
+        const resStore = eval(res)
+        assert.strictEqual(resStore.de.main.foo, undefined)
+        assertCommon(resStore)
+      })
+    })
+  })
```
Comment (rosskevin) [inline test/basename.spec.js]: Let's yank the include/exclude tests here, they are superfluous to this option, otherwise we are good to go

**Review by rosskevin — CHANGES_REQUESTED** [review]

**Pull request merged**